src/profile-logic/profile-data.js

```diff
@@ -391,10 +392,157 @@ function _createCallNodeInfoFromUnorderedComponents(
       stackIndexToCallNodeIndex: stackIndexToCallNodeIndex.map(
         (oldIndex) => oldIndexToNewIndex[oldIndex]
       ),
+      isInverted: false,
     };
   });
 }
 
+export function getInvertedCallNodeInfo(
+  thread: Thread,
+  defaultCategory: IndexIntoCategoryList
+): CallNodeInfo {
+  // We compute an inverted stack table, but we don't let it escape this function.
+  const {
+    invertedThread,
+    oldStackToNewStack: nonInvertedStackToInvertedStack,
+  } = _computeThreadWithInvertedStackTable(thread, defaultCategory);
+
+  // Create an inverted call node table based on the inverted stack table.
+  const {
+    callNodeTable,
+    stackIndexToCallNodeIndex: invertedStackIndexToCallNodeIndex,
+  } = getCallNodeInfo(
+    invertedThread.stackTable,
+    invertedThread.frameTable,
+    invertedThread.funcTable,
+    defaultCategory
+  );
+
+  // Create a mapping that maps a stack index from the non-inverted thread to
+  // its corresponding call node in the inverted tree.
+  const nonInvertedStackIndexToCallNodeIndex = new Int32Array(
+    thread.stackTable.length
+  );
+  for (
+    let nonInvertedStackIndex = 0;
+    nonInvertedStackIndex < nonInvertedStackIndexToCallNodeIndex.length;
+    nonInvertedStackIndex++
+  ) {
+    const invertedStackIndex = nonInvertedStackToInvertedStack.get(
+      nonInvertedStackIndex
+    );
+    if (invertedStackIndex === undefined) {
+      // This stack is not used as a self stack, only as a prefix stack.
+      // There is no inverted call node that corresponds to it.
```

Comment:
You're right; as written, the comment was not correct. I've changed it to:

```js
      // This stack is not used as a self stack, only as a prefix stack.
      // There may or may not be an inverted call node that corresponds to it,
      // but we haven't checked that and we don't need to know it.
      // nonInvertedStackIndexToCallNodeIndex only needs useful values for self stacks.
```

> After all, if we have A -> B -> C in the uniniverted call tree, we'll have C -> B -> A in the inverted call tree,

correct

> so A -> B is now C -> B

Oh, but if we wanted to put a value into `nonInvertedStackIndexToCallNodeIndex` for A -> B, it should be B -> A, not C -> B.

> (and if we have A -> B -> D too, then D -> B will match too).

It would "match" in some sense, but `nonInvertedStackIndexToCallNodeIndex` should be set up in such a way that the call paths of the key and the value are the same (just reversed).